plugins/quota/validate.go

```diff
@@ -5,34 +5,67 @@ import (
 
 	"github.com/gluster/glusterd2/glusterd2/volume"
 	"github.com/gluster/glusterd2/glusterd2/xlator"
+	"github.com/gluster/glusterd2/pkg/errors"
 )
 
 const (
-	quotaEnabledKey = "features.quota"
+	quotaEnabledKey = "quota.enable"
 )
 
 func validateOptions(v *volume.Volinfo, key string, value string) error {
 
-	if isQuotaEnabled(v) == false {
-		err := fmt.Errorf("Quota not enabled to set this value: '%s'", key)
+	// Check if volume is started
+	if v.State != volume.VolStarted {
+		err := errors.ErrVolNotStarted
 		return err
 	}
 
 	switch key {
-	case "deem-statfs":
+	case "enable":
+		switch value {
+		case "on":
+			if isQuotaEnabled(v) {
+				err := errors.ErrProcessAlreadyRunning
+				return err
+			}
+		case "off":
+			if !isQuotaEnabled(v) {
+				err := errors.ErrProcessNotRunning
+				return err
+			}
+		}
 		return nil
+	case "deem-statfs":
+		if isQuotaEnabled(v) == false {
+			err := fmt.Errorf("Quota not enabled to set this value: '%s'", key)
+			return err
+		}
 	case "hard-timeout":
-		return nil
+		if isQuotaEnabled(v) == false {
+			err := fmt.Errorf("Quota not enabled to set this value: '%s'", key)
+			return err
+		}
 	case "soft-timeout":
-		return nil
+		if isQuotaEnabled(v) == false {
+			err := fmt.Errorf("Quota not enabled to set this value: '%s'", key)
+			return err
+		}
 	case "alert-time":
-		return nil
+		if isQuotaEnabled(v) == false {
+			err := fmt.Errorf("Quota not enabled to set this value: '%s'", key)
+			return err
+		}
 	case "default-soft-limit":
-		return nil
+		if isQuotaEnabled(v) == false {
```

Comment:
There's a keyword in go called `fallthrough`. Please look it up. It can help you get rid of this repetitive code.